diff --git a/node_modules/@tumihub/react-native-thermal-receipt-printer/android/build.gradle b/node_modules/@tumihub/react-native-thermal-receipt-printer/android/build.gradle
index 3bd16c7..caed91a 100644
--- a/node_modules/@tumihub/react-native-thermal-receipt-printer/android/build.gradle
+++ b/node_modules/@tumihub/react-native-thermal-receipt-printer/android/build.gradle
@@ -1,8 +1,8 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion = 29
-    buildToolsVersion = "29.0.2"
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         minSdkVersion 16
