diff --git a/node_modules/react-native-bluetooth-state-manager/android/CMakeLists.txt b/node_modules/react-native-bluetooth-state-manager/android/CMakeLists.txt
index 789e494..d7cd6ae 100644
--- a/node_modules/react-native-bluetooth-state-manager/android/CMakeLists.txt
+++ b/node_modules/react-native-bluetooth-state-manager/android/CMakeLists.txt
@@ -5,6 +5,8 @@ set (PACKAGE_NAME BluetoothStateManager)
 set (CMAKE_VERBOSE_MAKEFILE ON)
 set (CMAKE_CXX_STANDARD 20)
 
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,max-page-size=16384")
+
 # Define C++ library and add all sources
 add_library(${PACKAGE_NAME} SHARED
         src/main/cpp/cpp-adapter.cpp
diff --git a/node_modules/react-native-bluetooth-state-manager/android/build.gradle b/node_modules/react-native-bluetooth-state-manager/android/build.gradle
index 96fccd6..5e3c875 100644
--- a/node_modules/react-native-bluetooth-state-manager/android/build.gradle
+++ b/node_modules/react-native-bluetooth-state-manager/android/build.gradle
@@ -51,9 +51,11 @@ android {
 
     externalNativeBuild {
       cmake {
-        cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
+        // Added -Wl,-z,max-page-size=16384 to cppFlags
+        cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all -Wl,-z,max-page-size=16384"
         arguments "-DANDROID_STL=c++_shared",
-          "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
+          "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
+          "-DANDROID_EXT_LINKER_FLAGS=-Wl,-z,max-page-size=16384" // Added this argument
         abiFilters (*reactNativeArchitectures())
       }
     }
